Replay's implementation

JiNT1 opened the pull request:
simple implementation without output's part(will be added later)

JiNT1 committed:
First replay's implementation

missing output's part

Aehronburn commented on src/targets/replay.cpp:
```diff
@@ -0,0 +1,100 @@
+	}
+	return movements;
+}
+std::map<char,unsigned int> board_rank
```
c'è una cosa simile in Controller. Si potrebbe spostare in Utilities?

milovangudelj replied:
Volendo, si. Riduce la possibilità di errore.

Aehronburn commented on src/targets/replay.cpp:
```diff
@@ -0,0 +1,100 @@
+	{'7',1},
+	{'8',0}
+};
+std::map<char,unsigned int> board_file
```
puoi sostituire con (char) - 'A'

JiNT1 committed:
Merge branch 'main' into Replay_implementation

JiNT1 committed:
Update replay.cpp

now replay.cpp should print board with a list of movement. Missing controller's list to print more information about board

JiNT1 committed:
Merge branch 'main' into Replay_implementation

JiNT1 committed:
Merge branch 'main' into Replay_implementation

JiNT1 committed:
Merge branch 'main' into Replay_implementation

JiNT1 committed:
Update replay.cpp

add method_v and method_f
add code for printing board every 1 sec

Aehronburn commented on src/targets/replay.cpp:
```diff
@@ -1,8 +1,134 @@
+		pos_whitespace = s.find(' ');
+		start_s = s.substr(0,pos_whitespace);
+		end_s = s.substr(pos_whitespace+1);
+		start_c = {(unsigned int)(8-(start_s.at(1)-'0')),(unsigned int)(start_s.at(0)-'A')};
```
prova con i cast stile c++

Aehronburn commented on src/targets/replay.cpp:
```diff
@@ -1,8 +1,134 @@
+	int pos_whitespace;
+	while(std::getline(in,s))
+	{
+		pos_whitespace = s.find(' ');
```
la lunghezza di una riga è costante?

JiNT1 replied:
si perchè vengono passate movimenti del tipo B2 B3,no?

Aehronburn replied:
secondo me puoi usare un metodo più efficiente di find

JiNT1 committed:
Merge branch 'main' into Replay_implementation

JiNT1 committed:
Update replay.cpp

changed c-style cast into c++-style;
update the cursor for printing

JiNT1 committed:
Update replay.cpp

milovangudelj committed:
Added `operator+` between string and color

milovangudelj committed:
Adde match result to `to_print`

milovangudelj committed:
Fixed cursor climbing up the window + mvmt to file

milovangudelj committed:
Merge branch 'Replay_implementation' of https://github.com/milovangudelj/scacchiera into Replay_implementation

milovangudelj committed:
Added invalid movement handling for `replay()`
